tests/flow/test_all_shortest_paths.py

```diff
@@ -0,0 +1,289 @@
+        expected_result = []
+        self.env.assertEqual(actual_result.result_set, expected_result)
+
+    def test02_all_shortest_paths(self):
+        # running against following graph
+        #
+        # (v1)-[:E]->(v2)-[:E]->(v3)-[:E]->(v4)
+        # (v1)-[:E]->(v5)-[:E2]->(v4)
+        # (v2)-[:E2]->(v4)
+
+        query = """MATCH (v1 {v: 1}), (v4 {v: 4})
+                   WITH v1, v4
+                   MATCH p = allShortestPaths((v1)-[*]->(v4))
+                   RETURN nodes(p) AS nodes
+                   ORDER BY nodes"""
+
+        actual_result = self.redis_graph.query(query)
+        # The 2 2-hop traversals should be found
+        expected_result = [[[self.v1, self.v2, self.v4]],
+                           [[self.v1, self.v5, self.v4]]]
+        self.env.assertEqual(actual_result.result_set, expected_result)
+
+        # Verify that a right-to-left traversal produces the same results
+        query = """MATCH (v1 {v: 1}), (v4 {v: 4})
+                   WITH v1, v4
+                   MATCH p = allShortestPaths((v4)<-[*]-(v1))
+                   RETURN nodes(p) AS nodes
+                   ORDER BY nodes"""
+
+        self.env.assertEqual(actual_result.result_set, expected_result)
+
+    def test03_all_shortest_min_hops_fail(self):
+        # running against following graph
+        #
+        # (v1)-[:E]->(v2)-[:E]->(v3)-[:E]->(v4)
+        # (v1)-[:E]->(v5)-[:E2]->(v4)
+        # (v2)-[:E2]->(v4)
+
+        query = """MATCH (v1 {v: 1}), (v4 {v: 4})
+                   WITH v1, v4
+                   MATCH p = allShortestPaths((v1)-[*3..]->(v4))
+                   RETURN nodes(p) AS nodes
+                   ORDER BY nodes"""
+
+        try:
+            self.redis_graph.query(query)
+            self.env.assertTrue(False)
+        except redis.exceptions.ResponseError as e:
+            self.env.assertIn("allShortestPaths(...) does not support a minimal length different from 1", str(e))
+
+    def test04_all_shortest_multiple_traversals(self):
+        # running against following graph
+        #
+        # (v1)-[:E]->(v2)-[:E]->(v3)-[:E]->(v4)
+        # (v1)-[:E]->(v5)-[:E2]->(v4)
+        # (v2)-[:E2]->(v4)
+
+        query = """MATCH (v1 {v: 1}), (v4 {v: 4})
+                   WITH v1, v4
+                   MATCH p = allShortestPaths((v1)-[:E]->(:L)-[*..]->(v4))
+                   RETURN nodes(p) AS nodes
+                   ORDER BY nodes"""
+
+        actual_result = self.redis_graph.query(query)
+        # The single 3-hop traversal should be found
+        expected_result = [[[self.v1, self.v2, self.v4]], 
+                           [[self.v1, self.v5, self.v4]]]
+        self.env.assertEqual(actual_result.result_set, expected_result)
+
+        # Verify that a right-to-left traversal produces the same results
+        query = """MATCH (v1 {v: 1}), (v4 {v: 4})
+                   WITH v1, v4
+                   MATCH p = allShortestPaths((v4)<-[*..]-(:L)<-[:E]-(v1))
```

Comment:
Question, do we utilise conditional traversal to see if `v1` is connected via `E` ?

Reply:
this is the plan
```
Results
    Sort
        Project
            Conditional Variable Length Traverse (Expand Into) | (anon_3:L)-[anon_4*1..INF]->(v4)
                Conditional Traverse | (v1)-[anon_2:E]->(anon_3:L)
                    Project
                        Cartesian Product
                            Filter
                                All Node Scan | (v1)
                            Filter
                                All Node Scan | (v4)
```